test/sql/optimizer_sql_test.cpp

```diff
@@ -124,17 +127,19 @@ class OptimizerSQLTests : public PelotonTest {
 };
 
 TEST_F(OptimizerSQLTests, SimpleSelectTest) {
+
   // Testing select star expression
   TestUtil("SELECT * from test", {"333", "22", "1", "2", "11", "0", "3", "33",
                                   "444", "4", "0", "555"},
            false);
-  
+
   // Something wrong with column property.
   string query = "SELECT b from test order by c";
-   
+
   // check for plan node type
   auto select_plan =
       TestingSQLUtil::GeneratePlanWithOptimizer(optimizer, query);
+  delete optimizer->txn;
```

Comment:
optimizer has member variable txn. If I don't delete it, the txn will be the previous txn, because in this test, optimizer is globe variable.